explict name for default network is unsupported in old docker-compose

roncohen opened the pull request:
this sets an explicit network name in each service

jalvz commented on scripts/compose.py:
```diff
@@ -234,6 +234,10 @@ def default_logging():
+    
+    @staticmethod
+    def default_networks():
+        return ["net1"]
```
maybe a more descriptive name? :)

roncohen replied:
good point 👍

graphaelli commented on scripts/compose.py:
```diff
@@ -1815,6 +1820,14 @@ def init_start_parser(self, parser, services, argv=None):
             default=DEFAULT_APM_SERVER_URL,
         )
 
+        parser.add_argument(
```
`--apm-server-secret-token` is already an option, should use that instead.

graphaelli replied:
note you'll need to merge master in following #292

roncohen replied:
thanks. I've reverted my changes. There's still the separate option for `--secret-token` for sourcemap uploads as it is. I decided not to try and change that in this PR.

roncohen commented:
* Changed sourcemap upload URL to work with 7.0. 
* Discovered that Curl would silently fail if the wrong token or wrong URL was used. I changed some of the arguments to curl to make it fail loudly now if upload doesn't work. The regular case (correct token/url) is slightly more noisy, but i think it's worths it: 

Now, correct token:
```
 $ ./scripts/compose.py upload-sourcemap --secret-token abc
Uploading /Users/ron/Projects/elastic/apm-integration-testing/docker/opbeans/node/sourcemaps/main.a69bd995.js.map for client version 1.0.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2383k    0     0  100 2383k      0  15.0M --:--:-- --:--:-- --:--:-- 15.1M
```
Incorrect token:
```
$ ./scripts/compose.py upload-sourcemap --secret-token asd
Uploading /Users/ron/Projects/elastic/apm-integration-testing/docker/opbeans/node/sourcemaps/main.a69bd995.js.map for client version 1.0.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0 2383k    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 401 Unauthorized
Traceback (most recent call last):
  File "./scripts/compose.py", line 2235, in <module>
    main()
  File "./scripts/compose.py", line 2231, in main
    setup()
  File "./scripts/compose.py", line 1687, in __call__
    self.args.func()
  File "./scripts/compose.py", line 2083, in upload_sourcemaps_handler
    subprocess.check_output(cmd, shell=True).decode('utf8').strip()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command 'docker run --rm --network apm-integration-testing -v /Users/ron/Projects/elastic/apm-integration-testing/docker/opbeans/node/sourcemaps/main.a69bd995.js.map:/tmp/sourcemap centos:7 curl -f -X POST -F service_name="client" -F service_version="1.0.0" -F bundle_filepath="http://opbeans-node:3000/static/js/main.a69bd995.js.map" -F sourcemap=@/tmp/sourcemap -H "Authorization: Bearer asd" http://apm-server:8200/assets/v1/sourcemaps' returned non-zero exit status 22
```

roncohen commented:
even setting `{"name": "apm-integration-testing"}` doesn't work in older docker-compose versions unfortunately, so i've completely removed it

roncohen commented:
not sure if we want to merge this. The network name will now be set to`<parent-directory>_default`

kuisathaverat commented:
@roncohen Is this PR still useful?
